src/Files.App/Actions/Content/Background/SetAsWallpaperBackgroundAction.cs

```diff
@@ -18,20 +18,21 @@ public override RichGlyph Glyph
 
 		public override bool IsExecutable =>
 			base.IsExecutable &&
-			context.SelectedItem is not null;
+			ContentPageContext.SelectedItem is not null;
 
 		public override Task ExecuteAsync(object? parameter = null)
 		{
-			if (context.SelectedItem is null)
+			if (ContentPageContext.SelectedItem is null)
 				return Task.CompletedTask;
 
 			try
 			{
-				WindowsWallpaperService.SetDesktopWallpaper(context.SelectedItem.ItemPath);
+				WindowsWallpaperService.SetDesktopWallpaper(ContentPageContext.SelectedItem.ItemPath);
 			}
 			catch (Exception ex)
 			{
 				ShowErrorDialog(ex.Message);
+				App.Logger.LogWarning(ex, ex.Message);
```

Comment:
I don't think logging helps us because there isn't anything to do for this type of exception. The best thing is to notify the user and prevent a crash.

Reply:
It's not really relevant to users, the reason we have a dialog in the first place is so that the user is informed that the action didn't complete.

Reply:
It's common to include HResult error code into user-facing dialog though.
I'll remove it then.

Reply:
It's okay if we think it'll be helpful